From 2c188634e90ad60dce9e1ff734b7122c1e8d7b69 Mon Sep 17 00:00:00 2001
From: Mao Minkai <maominkai@wxiat.com>
Date: Tue, 7 Jun 2022 17:41:38 +0800
Subject: [PATCH] sw64: print correct initrd address
Patch-mainline: NO, OTHERS
References: openEuler-22.03-LTS
Modified-by-SEL: No


Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I56TYY

--------------------------------

Print correct initrd address if move_initrd() failed.

Signed-off-by: Mao Minkai <maominkai@wxiat.com>

Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Acked-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/kernel/setup.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/arch/sw_64/kernel/setup.c b/arch/sw_64/kernel/setup.c
index 250a45ee56f9..2784ea5aa0ae 100644
--- a/arch/sw_64/kernel/setup.c
+++ b/arch/sw_64/kernel/setup.c
@@ -407,10 +407,12 @@ void __init process_memmap(void)
 			break;
 		case memmap_initrd:
 			if (!memmap_range_valid(base, size)) {
+				phys_addr_t old_base = base;
+
 				base = (unsigned long) move_initrd(memblock_end_of_DRAM());
 				if (!base) {
 					pr_err("initrd memmap region [mem %#018llx-%#018llx] extends beyond end of memory (%#018llx)\n",
-							base, base + size - 1, memblock_end_of_DRAM());
+							old_base, old_base + size - 1, memblock_end_of_DRAM());
 				} else {
 					memmap_map[i].addr = base;
 					pr_info("initrd memmap region [mem %#018llx-%#018llx]\n",
-- 
2.34.1

